src/lsm/compaction.zig

```diff
@@ -148,7 +148,7 @@ pub fn CompactionType(
         );
 
         const MergeStreamSelector = struct {
-            fn peek(compaction: *const Compaction, stream_id: u32) ?Key {
+            fn peek(compaction: *const Compaction, stream_id: u32) error{Empty, Buffering}!Key {
```

Comment:
Rather than `Buffering` could we call it `Drained`? "Buffering" isn't necessarily a correct description, since there might not be an I/O in progress right now. (Once we have better compaction pacing, this will be true if it isn't already the case).